raiden/tests/unit/transfer/mediated_transfer/test_mediation_fee.py

```diff
@@ -434,6 +430,69 @@ def test_fee_round_trip(flat_fee, prop_fee, imbalance_fee, amount, balance1, bal
     assert amount_with_margin_after_fees >= amount
 
 
+@example(flat_fee=0, prop_fee=0, imbalance_fee=1277, amount=1, balance1=33, balance2=481)
+@given(
+    integers(min_value=0, max_value=100),
+    integers(min_value=0, max_value=10_000),
+    integers(min_value=0, max_value=50_000),
+    integers(min_value=1, max_value=90_000_000_000_000_000_000),
+    integers(min_value=1, max_value=100_000_000_000_000_000_000),
+    integers(min_value=1, max_value=100_000_000_000_000_000_000),
+)
+def test_fee_add_remove_invariant(flat_fee, prop_fee, imbalance_fee, amount, balance1, balance2):
+    """ First adding and then removing fees must yield the original value """
+    # Find examples where there is a reasonable chance of succeeding
+    amount = int(min(amount, balance1 * 0.95 - 1, balance2 * 0.95 - 1))
+    assume(amount > 0)
+
+    total_balance = TokenAmount(100_000_000_000_000_000_000)
+    prop_fee_per_channel = ppm_fee_per_channel(ProportionalFeeAmount(prop_fee))
+    imbalance_fee = calculate_imbalance_fees(
+        channel_capacity=total_balance,
+        proportional_imbalance_fee=ProportionalFeeAmount(imbalance_fee),
+    )
+    fee_schedule = FeeScheduleState(
+        cap_fees=False,
+        flat=FeeAmount(flat_fee),
+        proportional=prop_fee_per_channel,
+        imbalance_penalty=imbalance_fee,
+    )
+    channel_in = factories.create(
+        NettingChannelStateProperties(
+            our_state=NettingChannelEndStateProperties(balance=total_balance - balance1),
+            partner_state=NettingChannelEndStateProperties(balance=balance1),
+            fee_schedule=fee_schedule,
+        )
+    )
+    channel_out = factories.create(
+        NettingChannelStateProperties(
+            our_state=NettingChannelEndStateProperties(balance=balance2),
+            partner_state=NettingChannelEndStateProperties(balance=total_balance - balance2),
+            fee_schedule=fee_schedule,
+        )
+    )
+
+    amount_with_fees = get_amount_with_fees(
+        amount_without_fees=amount,
+        schedule_in=channel_in.fee_schedule,
+        schedule_out=channel_out.fee_schedule,
+        receivable_amount=balance1,
+        balance_in=total_balance - balance1,
+        balance_out=balance2,
+    )
+    assume(amount_with_fees)
+    assert amount_with_fees
+    amount_without_fees = get_amount_without_fees(
+        amount_with_fees=amount_with_fees, channel_in=channel_in, channel_out=channel_out
+    )
+    assume(amount_without_fees)
+    fee_in = float(channel_in.fee_schedule.fee(total_balance - balance1, amount_with_fees))
+    fee_out = float(channel_out.fee_schedule.fee(balance2, -amount))
+    print(f"amount: {amount} with_fees: {amount_with_fees} without_fees: {amount_without_fees}")
```

Comment:
I guess those print statements were for your own debugging and can now be removed?

Reply:
They are nice to have when a test fails. So I would keep them in instead of removing them and adding them temporarily back when the test is causing problems. I can remove them if that's what we want to do for consistency or if there are any downsides to keeping them in.

Reply:
Why not assert on the amounts and then if they are not as expected have the strings that you print there at the assertion failure?

> . I can remove them if that's what we want to do for consistency or if there are any downsides to keeping them in.

In general we don't leave print statements in the tests.

Reply:
> Why not assert on the amounts and then if they are not as expected have the strings that you print there at the assertion failure?

Because we don't know which values these variables should have. But when the sum of them does not match the expectation, it is interesting to know the summands to find out what went wrong. With the default behaviour of pytest (capture output and only show output on test failure), this is a pleasant way to lightweight debugging logs in tests.

I'll remove this for consistency with the other raiden tests.